client-app/ui-kit/components/atoms/typography/vc-typography.vue

```diff
@@ -1,93 +1,167 @@
 <template>
   <component
     :is="tag"
-    :class="['vc-typography', `vc-typography--variant--${variant}`, `vc-typography--weight--${weight}`]"
+    :class="[
+      'vc-typography',
+      `vc-typography--variant--${_variant}`,
+      `vc-typography--weight--${_weight}`,
+      `vc-typography--case--${_textTransform}`,
+      {
+        'vc-typography--truncate': truncate,
+      },
+    ]"
   >
     <slot />
   </component>
 </template>
 
 <script setup lang="ts">
-interface Props {
+import { computed } from "vue";
+
+interface IProps {
   tag?: string;
-  variant?: "h1" | "h2" | "h3" | "h4" | "h5" | "h6" | "large" | "base" | "medium" | "small";
-  weight?: "normal" | "semibold" | "bold" | "extrabold";
+  variant?: "h1" | "h2" | "h3" | "h4" | "h5" | "h6" | "base";
+  weight?: "thin" | "light" | "normal" | "bold" | "black";
+  textTransform?: "uppercase" | "lowercase" | "capitalize" | "none";
+  truncate?: boolean;
 }
 
-withDefaults(defineProps<Props>(), {
+const props = withDefaults(defineProps<IProps>(), {
   tag: "p",
-  variant: "base",
-  weight: "normal",
+  align: "left",
+});
+
+const isHeader = computed(() => /^(h[1-6])$/.test(props.tag));
+
+const _variant = computed(() => {
+  if (props.variant) {
+    return props.variant;
+  }
+
+  if (isHeader.value) {
+    return props.tag;
+  }
+
+  return "base";
+});
+
+const _weight = computed(() => {
+  if (props.weight) {
+    return props.weight;
+  }
+
+  if (isHeader.value) {
+    return "bold";
+  }
+
+  return "normal";
+});
+
+const _textTransform = computed(() => {
+  if (props.textTransform) {
+    return props.textTransform;
+  }
+
+  if (isHeader.value) {
+    return "uppercase";
+  }
+
+  return "none";
 });
```

Comment:
I'm not sure this is good for extensibility, May be it's better to do it in styles instead?